BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
+{
+    [JsonPropertyName("currencyCode")] public string CurrencyCode { get; set; }
+    [JsonPropertyName("rate")] public float RateValue { get; set; }
+}
+
+internal class ExchangeRateService
+{
+    private const string MainCurrenciesUrl = "https://api.cnb.cz/cnbapi/exrates/daily";
+
+    public IEnumerable<Rate> GetExchangeRates(List<string> currencies)
+    {
+        HttpClient client = new HttpClient();
+			
+        var data = client.GetStringAsync(MainCurrenciesUrl).Result;
+        var response = JsonSerializer.Deserialize<CnbRatesResponse>(data);
+
+        return response.Rates.Select(rate => rate).Where(rate => currencies.Contains(rate.CurrencyCode));
```

Comment:
This line is functional, but based on the earlier suggestions:

1. We could move the filtering logic into a dedicated method like FilterRates.
2. Instead of exposing the Rate domain model, consider returning a RateDTO here to avoid leaking domain logic.
3. Finally, ensure that the currencies list is validated before this step to avoid unexpected results.

If you've followed along on all the other comments, you should have a nicer return than this here 👍